src/serverlist.cpp

```diff
@@ -490,6 +541,105 @@ void CServerListManager::CentralServerQueryServerList ( const CHostAddress& Inet
     }
 }
 
+int CServerListManager::IndexOf ( CHostAddress haSearchTerm )
+{
+    // Find the server in the list. The very first list entry
+    // per definition is the directory server
+    // (i.e., this server).
+    for ( int iIdx = ServerList.size() - 1; iIdx > 0; iIdx-- )
+    {
+        if ( ServerList[iIdx].HostAddr == haSearchTerm )
+        {
+            return iIdx;
+        }
+    }
+    return INVALID_INDEX;
+}
+
+void CServerListManager::CentralServerLoadServerList ( const QString strServerList )
+{
+    QFile file ( strServerList );
+
+    if ( !file.open ( QIODevice::ReadWrite | QIODevice::Text ) )
+    {
+        qWarning() << qUtf8Printable ( QString ( "Could not open '%1' for read/write" ).arg ( ServerListFileName ) );
```

Comment:
```suggestion
        qWarning() << qUtf8Printable ( QString ( "Could not open '%1' for read/write. Please check if Jamulus has read and write permissions and if your disk is full." ).arg ( ServerListFileName ) );
```

Goes with https://github.com/jamulussoftware/jamulus/discussions/1568